frame/composable-traits/src/lending/mod.rs

```diff
@@ -238,45 +355,119 @@ pub trait Lending: DeFiEngine {
 
 	/// utilization_ratio = total_borrows / (total_cash + total_borrows).
 	/// utilization ratio is 0 when there are no borrows.
-	fn calc_utilization_ratio(
+	fn calculate_utilization_ratio(
 		cash: Self::Balance,
 		borrows: Self::Balance,
 	) -> Result<Percent, DispatchError>;
 
-	/// Borrow asset amount account should repay to be debt free for specific market pair.
-	/// Calculate account's borrow balance using the borrow index at the start of block time.
+	/// The amount of *borrow asset* debt remaining for the account in the specified market,
+	/// including accrued interest.
+	///
+	/// Could also be thought of as the amount of *borrow asset* the account must repay to be
+	/// totally debt free in the specified market.
+	///
+	/// Calculates the account's borrow balance using the borrow index at the start of block time.
+	///
 	/// ```python
 	/// new_borrow_balance = principal * (market_borrow_index / borrower_borrow_index)
 	/// ```
-	fn borrow_balance_current(
+	fn total_debt_with_interest(
 		market_id: &Self::MarketId,
 		account: &Self::AccountId,
-	) -> Result<Option<BorrowAmountOf<Self>>, DispatchError>;
+	) -> Result<TotalDebtWithInterest<BorrowAmountOf<Self>>, DispatchError>;
 
 	fn collateral_of_account(
 		market_id: &Self::MarketId,
 		account: &Self::AccountId,
 	) -> Result<CollateralLpAmountOf<Self>, DispatchError>;
 
 	/// Borrower shouldn't borrow more than his total collateral value
+	///
+	/// The amount of collateral that would be required in order to borrow `borrow_amount` of borrow
+	/// asset.
+	///
+	/// Can be thought of as the "inverse" of [`Lending::get_borrow_limit`], in that
+	/// `get_borrow_limit` returns the maximum amount borrowable with the *current* collateral,
+	/// while `collateral_required` returns the amount of collateral asset that would be needed to
+	/// borrow the specified amount.
 	fn collateral_required(
 		market_id: &Self::MarketId,
 		borrow_amount: Self::Balance,
 	) -> Result<Self::Balance, DispatchError>;
 
-	/// Returns the borrow limit for an account in `Oracle` price.
-	/// Calculation uses indexes from start of block time.
-	/// Depends on overall collateral put by user into vault.
-	/// This borrow limit of specific user, depends only on prices and users collateral, not on
-	/// state of vault.
-	/// ```python
-	/// collateral_balance * collateral_price / collateral_factor - borrower_balance_with_interest * borrow_price
+	/// Returns the "borrow limit" for an account in `Oracle` price, i.e. the maximum amount an
+	/// account can borrow before going under-collateralized.
+	///
+	/// REVIEW: What?
+	/// The calculation uses indexes from start of block time.
+	///
+	/// The borrow limit is only affected by the prices of the assets and the amount of collateral
+	/// deopsited by the account, and is *specific to this account*. The state of the vault is not
+	/// relevant for this calculation.
+	///
+	/// The calculation is as follows, broken up for clarity:
+	///
+	/// ```ignore
```

Comment:
i liked python more, cause it allows to cpy paste code in python and get all what you want, formulas, graphs, examples. unlike rust or haskell, default numbers are bigdecimals and there is no types to type - which makes python 1 to 1 with math.